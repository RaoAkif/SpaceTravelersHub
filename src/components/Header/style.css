.navbar-brand {
  padding-left: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
  text-decoration: none;
  font-size: 20px;
  font-weight: bold;
  color: #00d7c6;
}

.logo {
  width: 50px;
  height: 50px;
}

.logo-text {
  margin-left: 10px;
}

.me-auto {
  margin: 2px;
  margin-left: 35rem;
}

.me-auto > * {
  text-decoration: none;
  color: grey;
  font-size: 16px;
}

.nav-link {
  margin: 1px;
}

.nav {
  margin-bottom: 25px;
}

.nav a {
  border-radius: 4px;
  padding: 6px 10px;
  margin-right: 10px;
}

.nav a.active {
  color: #fff;
  background: #7600dc;
}

.nav a.inactive {
  font-style: italic;
}
