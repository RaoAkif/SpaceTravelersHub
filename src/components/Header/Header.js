import React from 'react';
import { NavLink } from 'react-router-dom';
import { Navbar, Nav, Container } from 'react-bootstrap';
import logo from '../../assets/images/logo.png';
import './style.css';

function Header() {
  return (
    <div>
      <Navbar bg="dark" variant="dark" expand="md" className="Nav-bar">
        <Container>
          <Navbar.Brand href="#home">
            <img
              alt=""
              src={logo}
              width="30"
              height="30"
              className="logo d-inline-block align-top"
            />
            {' '}
            Space Traverlers Hub
          </Navbar.Brand>
          <Nav className="me-auto">
            <NavLink className="nav-link" style={({ isActive }) => ({ color: isActive ? '#fff' : '#545e6f', background: isActive ? '#00d7c6' : '#f0f0f0' })} to="/">Rockets</NavLink>
            <NavLink className="nav-link" style={({ isActive }) => ({ color: isActive ? '#fff' : '#545e6f', background: isActive ? '#00d7c6' : '#f0f0f0' })} to="/missions">Missions</NavLink>
            <NavLink className="nav-link" style={({ isActive }) => ({ color: isActive ? '#fff' : '#545e6f', background: isActive ? '#00d7c6' : '#f0f0f0' })} to="/myprofile">My Profile</NavLink>
          </Nav>
        </Container>
      </Navbar>
    </div>
  );
}

export default Header;
