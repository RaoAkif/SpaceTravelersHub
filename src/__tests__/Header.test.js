import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Header from '../components/Header/Header';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
