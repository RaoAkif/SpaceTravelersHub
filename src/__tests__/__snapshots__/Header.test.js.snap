// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="Nav-bar navbar navbar-expand navbar-dark bg-dark"
  >
    <div
      className="container"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": "80px",
          "justifyContent": "space-between",
        }
      }
    >
      <a
        className="navbar-brand"
        href="#home"
      >
        <img
          alt=""
          className="logo d-inline-block align-top"
          height="30"
          src="logo.png"
          width="30"
        />
         
        <span
          className="logo-text"
        >
          Space Traverlers Hub
        </span>
      </a>
      <div
        className="me-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "background": "#00d7c6",
              "color": "#fff",
            }
          }
        >
          Rockets
        </a>
        <a
          className="nav-link"
          href="/missions"
          onClick={[Function]}
          style={
            Object {
              "background": "#f0f0f0",
              "color": "#545e6f",
            }
          }
        >
          Missions
        </a>
        <a
          className="nav-link"
          href="/myprofile"
          onClick={[Function]}
          style={
            Object {
              "background": "#f0f0f0",
              "color": "#545e6f",
            }
          }
        >
          My Profile
        </a>
      </div>
    </div>
  </nav>
</div>
`;
